test/unity/toolkits/drawing_classifier/test_dc_serialization.cxx

```diff
@@ -174,8 +179,140 @@ BOOST_AUTO_TEST_CASE(test_export_coreml) {
                    target + "Probability");
 }
 
+BOOST_AUTO_TEST_CASE(test_load_version) {
+  turi::drawing_classifier::drawing_classifier dc;
+  TS_ASSERT_EQUALS(dc.get_version(), dc.DRAWING_CLASSIFIER_VERSION);
+}
+
+
 BOOST_AUTO_TEST_CASE(test_save_load) {
-  // TODO
+  auto remove_if_exist =
+      [](const char* fname) {
+        std::FILE* fp = nullptr;
+        if ((fp = std::fopen(fname, "r")) != NULL) {
+          if (std::fclose(fp) != 0) {
+            std::stringstream ss;
+            ss << "cannot close file: " << fname << std::endl;
+            throw(turi::error::io_error(ss.str()));
+          }
+
+          if (std::remove(fname) != 0) {
+            std::stringstream ss;
+            ss << "cannot remove file: " << fname << std::endl;
+            throw(turi::error::io_error(ss.str()));
+          }
+        }
+      };
+
+  /**
+   * This is a hack because nn_spec_ is a private member. Thus, I can't
+   * compare it directly after loading from file (load_version call).
+   *
+   * The solution is to call `load_version` first and write serialized
+   * `nn_spec_` to disk. In this way, I can indirectly verify the `nn_spec_`
+   * successively loads all params from the source model spec file.
```

Comment:
Could we instead just add a protected getter for `nn_spec_`?

The other thing we can do is implement the core logic for save and load as free functions, as we did with OD.

Reply:
should we expose the `nn_spec_` to user? or I can have return a copy to user. Either way.

Reply:
I'm suggesting that `drawing_classifier` expose a read-only view to subclasses, then the testing subclass can expose it publicly to tests